source/AI.cpp

```diff
@@ -2594,6 +2639,113 @@ bool AI::DoCloak(Ship &ship, Command &command)
 
 
 
+bool AI::DoRoving(Ship &ship, Command &command, unsigned int rovingShipCount)
+{	
+	const Point target = ship.GetTargetPosition();
+	const auto v = ship.MaxVelocity();
+	if(!target || MoveTo(ship, command, target, Point(), v, v))
+	{
+		Point newTarget = Angle::Random().Unit() * Random::Real() * MAX_DISTANCE_FROM_CENTER;
+		ship.SetTargetPosition(newTarget);
+	}
+	
+	const System *system = ship.GetSystem();
+	if(rovingShipCount > 9 && system)
+		return DoBored(ship, command, system);
+	
+	return false;
+}
+
+
+
+bool AI::DoEvasive(Ship &ship, Command &command)
+{
+	// Did I have to move my targetPosition?
+	bool movedTarget = false;
+	
+	// First, figure out which ships are not of the same government. Ignore disabled/destroyed ships.
+	vector<shared_ptr<Ship>> enemies;
+	for(auto otherShip : GetShipsList(ship, false))
+		if(otherShip->GetGovernment() != ship.GetGovernment())
+			enemies.push_back(make_shared<Ship>(*otherShip));
+	
+	// Loop through enemies and figure out their range and how to evade them.
+	vector<Point> paths;
+	double maxRange = 0.;
+	const Point pos = ship.Position();
+	for(const auto &enemy : enemies)
+	{
+		// Ignore disabled/destroyed ships.
+		if(enemy->IsDisabled() || enemy->IsDestroyed())
```

Comment:
```suggestion
	// Loop through enemies and figure out their range and how to evade them.
	vector<Point> paths;
	double maxRange = 0.;
	const Point pos = ship.Position();
	for(const auto &otherShip : GetShipsList(ship, false))
		if(otherShip->GetGovernment() == ship.GetGovernment() ||
				(enemy->IsDestroyed() || enemy->IsDisabled()))
			continue
```
We don't need the vector